packages/flutter_tools/lib/src/macos/build_macos.dart

```diff
@@ -243,6 +249,23 @@ Future<void> buildMacOS({
   ));
 }
 
+List<String> _computeDestinations(DarwinArch? darwinArch, String? configuration) {
+  final DarwinArch currentArch = switch (globals.os.hostPlatform) {
+    HostPlatform.darwin_arm64 => DarwinArch.arm64,
+    HostPlatform.darwin_x64 => DarwinArch.x86_64,
+    _ => throw ArgumentError(),
+  };
+
+  if (darwinArch == null) {
+    if (configuration == 'Debug') {
+      return <String>['-destination', 'platform=macOS,arch=${currentArch.name}'];
```

Comment:
I also wanted to get rid of bothersome warning when building in debug mode, which looks like this:
```
Launching lib/main.dart on macOS in debug mode...
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:arm64, id:00006000-000849E41162801E, name:My Mac }
{ platform:macOS, arch:x86_64, id:00006000-000849E41162801E, name:My Mac }
```

It happens because of `ONLY_ACTIVE_ARCH=YES` + `-destination 'platform=macOS'`: on Macs with Rosetta installed, both arches can be used as active. Here we narrow down destinations to use only current arch, gathered by `globals.os.hostPlatform`.

If it is not desired behaviour, or code becomes too cumbersome, I don't mind to undo this